ui/apps/platform/src/Containers/NetworkGraph/components/DeploymentSelector.tsx

```diff
@@ -67,47 +62,99 @@ function DeploymentSelector({
         setSearchFilter(modifiedSearchObject);
     };
 
-    const deploymentSelectOptions: JSX.Element[] = deploymentsByNamespace.map((namespace) => (
-        <SelectGroup label={namespace.metadata.name} key={namespace.metadata.id}>
-            {namespace.deployments.map((deployment) => (
-                <SelectOption key={deployment.id} value={deployment.name}>
+    let deploymentSelectMenuItems = deploymentsByNamespace.map((namespace) => {
+        const menuItems = namespace.deployments
+            .filter((deployment) =>
+                deployment.name.toLowerCase().includes(input.toString().toLowerCase())
+            )
+            .map((deployment) => (
+                <MenuItem
+                    hasCheck
+                    itemId={deployment.name}
+                    isSelected={selectedDeployments.includes(deployment.name)}
+                >
                     <span>
-                        <DeploymentIcon /> {deployment.name}
+                        <DeploymentIcon />
+                        <span className="pf-u-mx-xs" data-testid="deployment-name">
+                            {deployment.name}
+                        </span>
                     </span>
-                </SelectOption>
-            ))}
-        </SelectGroup>
-    ));
+                </MenuItem>
+            ));
+        if (menuItems.length === 0) {
+            return null;
+        }
```

Comment:
suggestion: If we are controlling the content anyway, we could substitute an unselectable "No deployments in this namespace match" when this happens.

Reply:
Are you saying to still show the namespace group text and just show the text below that with that message?

<img width="269" alt="Screenshot 2023-03-29 at 11 39 34 AM" src="https://user-images.githubusercontent.com/4805485/228636185-0d70b53d-6c38-46b5-86dc-1a1e01423c99.png">

I feel like it creates a bit of noise, especially if we have a lot of namespaces. This is what my changes will show in this PR

<img width="234" alt="Screenshot 2023-03-29 at 11 41 12 AM" src="https://user-images.githubusercontent.com/4805485/228636625-ff6cfac7-e3d4-4c30-8f54-5ee7ef63c9a7.png">

@zhenpesky maybe you might have opinions about this. Right now we currently show something like this

<img width="216" alt="Screenshot 2023-03-29 at 11 41 56 AM" src="https://user-images.githubusercontent.com/4805485/228636730-c756b630-cbf2-4878-98c8-b4019fef2789.png">

Reply:
For the message per namespace, I was thinking about situations like this one I just caused in stagingdb.
Notice the one NS has no matches, while the other still does. Maybe a full message for each NS would be too business--especially if there were 3 or more namespaces--so maybe just a dash `—`.

I agree that if there are no matches at all, it would be better to short-circuit to your one "No deployments found" message.

Reply:
I like the dash idea for the use case when there is a deployment found in one of the namespaces while other namespaces have zero deployments, we would just use a dash to represent "no deployment found in this namespace".

Reply:
Like this? @vjwilson @zhenpesky 

<img width="247" alt="Screenshot 2023-03-29 at 5 14 48 PM" src="https://user-images.githubusercontent.com/4805485/228695970-de03498e-b2da-457c-90a9-709d2203bf28.png">

Reply:
It looks a little odd to me, but if UX thinks its good I'll make the change

Reply:
I think if there are no deployments in _any_ namespace, then your single "No deployments found" message should replace the NS list with all dashes.